packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -167,12 +168,19 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
         LabelDescription labelDescription = nodeDescription.getLabelDescription();
         nodeVariableManager.put(LabelDescription.OWNER_ID, nodeId);
 
-        LabelType nodeLabelType = LabelType.INSIDE_CENTER;
-        if (containmentKind == NodeContainmentKind.BORDER_NODE) {
-            nodeLabelType = LabelType.OUTSIDE;
-        } else if (NodeType.NODE_IMAGE.equals(type)) {
-            nodeLabelType = LabelType.OUTSIDE_CENTER;
-        }
+        LabelType nodeLabelType = optionalPreviousNode.map(n -> LabelType.valueOfByValue(n.getLabel().getType())).orElseGet(() -> {
+            LabelType defaultLabeltype = null;
+            if (containmentKind == NodeContainmentKind.BORDER_NODE) {
+                defaultLabeltype = LabelType.OUTSIDE;
+            } else if (NodeType.NODE_IMAGE.equals(type)) {
+                defaultLabeltype = LabelType.OUTSIDE_CENTER;
+            } else if (NodeType.NODE_RECTANGLE.equals(type) && !this.hasHeader(style)) {
+                defaultLabeltype = LabelType.INSIDE_V_TOP_H_CENTER;
+            } else {
+                defaultLabeltype = LabelType.INSIDE_CENTER;
+            }
```

Comment:
Why separate two different label types for those use cases?

Reply:
`INSIDE_CENTER` is used for node parametric svgs, node icon labels, and node rectangles with header.
Then it is used by the frontend in `DependencyInjection.ts`. It allows to make differences between `InsideLabelView.tsx` and `LabelView.tsx`.